test/metabase/util/queue_test.clj

```diff
@@ -83,3 +83,20 @@
 
     (testing "The realtime events are processed in order"
       (mt/ordered-subset? realtime-events processed))))
+
+(deftest delay-queue-test
+  (let [q (queue/delay-queue)]
+    (dotimes [_ 5]
+      (queue/put-with-delay! q (+ 100 (* 100 (Math/random))) (System/nanoTime)))
+    (is (empty? (queue/take-delayed-batch! q 3)))
+    (is (nil? (queue/take-delayed-batch! q 3 90)))
+    (is (< 0 (count (queue/take-delayed-batch! q 3 110))))
+    (is (> 4 (count (queue/take-delayed-batch! q 3))))
+    (is (empty? (queue/take-delayed-batch! q 3))))
+
+  ;; TODO
```

Comment:
Made a note to come back and do "hardcore" thread safety checks later in the project, don't want to hold up getting base functionality in.